File: backend/src/modules/acc/dto/query-acc.dto.ts
import { IsOptional, IsString, IsDateString } from 'class-validator';
import { ApiPropertyOptional } from '@nestjs/swagger';
import { PaginationDto } from '../../../common/dto/pagination.dto';

export class QueryAccDto extends PaginationDto {
  @ApiPropertyOptional({ description: '来源用户' })
  @IsOptional()
  @IsString()
  srcUser?: string;

  @ApiPropertyOptional({ description: '目标用户' })
  @IsOptional()
  @IsString()
  dstUser?: string;

  @ApiPropertyOptional({ description: 'Call ID' })
  @IsOptional()
  @IsString()
  callid?: string;

  @ApiPropertyOptional({ description: '开始时间' })
  @IsOptional()
  @IsDateString()
  startTime?: string;

  @ApiPropertyOptional({ description: '结束时间' })
  @IsOptional()
  @IsDateString()
  endTime?: string;

  @ApiPropertyOptional({ description: 'SIP 响应码' })
  @IsOptional()
  @IsString()
  sipCode?: string;
}

File: kam-admin-module-monitor/src/main/java/io/kamailio/admin/modules/acc/dto/QueryAccDto.java
package io.kamailio.admin.modules.acc.dto;

import io.kamailio.admin.common.dto.PaginationDto;
import org.springframework.lang.Nullable;

public class QueryAccDto extends PaginationDto {
    @Nullable private String srcUser;
    @Nullable private String dstUser;
    @Nullable private String callid;
    @Nullable private String startTime;
    @Nullable private String endTime;
    @Nullable private String sipCode;

    public String getSrcUser() { return srcUser; }
    public void setSrcUser(String srcUser) { this.srcUser = srcUser; }
    public String getDstUser() { return dstUser; }
    public void setDstUser(String dstUser) { this.dstUser = dstUser; }
    public String getCallid() { return callid; }
    public void setCallid(String callid) { this.callid = callid; }
    public String getStartTime() { return startTime; }
    public void setStartTime(String startTime) { this.startTime = startTime; }
    public String getEndTime() { return endTime; }
    public void setEndTime(String endTime) { this.endTime = endTime; }
    public String getSipCode() { return sipCode; }
    public void setSipCode(String sipCode) { this.sipCode = sipCode; }
}
